pkg/app/piped/platformprovider/ecs/client.go

```diff
@@ -432,11 +456,45 @@ func (c *client) getLoadBalancerArn(ctx context.Context, targetGroupArn string)
 	return output.TargetGroups[0].LoadBalancerArns[0], nil
 }
 
-func (c *client) ModifyListeners(ctx context.Context, listenerArns []string, routingTrafficCfg RoutingTrafficConfig) error {
+func (c *client) ModifyListenerOrRule(ctx context.Context, listenerArns []string, listenerRuleArns []string, routingTrafficCfg RoutingTrafficConfig) error {
 	if len(routingTrafficCfg) != 2 {
 		return fmt.Errorf("invalid listener configuration: requires 2 target groups")
 	}
 
+	if len(listenerRuleArns) > 0 {
+		modifyListenerRule := func(ctx context.Context, listenerRuleArn string) error {
+			input := &elasticloadbalancingv2.ModifyRuleInput{
+				RuleArn: aws.String(listenerRuleArn),
+				Actions: []elbtypes.Action{
+					{
+						Type: elbtypes.ActionTypeEnumForward,
+						ForwardConfig: &elbtypes.ForwardActionConfig{
+							TargetGroups: []elbtypes.TargetGroupTuple{
+								{
+									TargetGroupArn: aws.String(routingTrafficCfg[0].TargetGroupArn),
+									Weight:         aws.Int32(int32(routingTrafficCfg[0].Weight)),
+								},
+								{
+									TargetGroupArn: aws.String(routingTrafficCfg[1].TargetGroupArn),
+									Weight:         aws.Int32(int32(routingTrafficCfg[1].Weight)),
+								},
```

Comment:
Will this always change a rule to `ForwardAction` type, which forwards traffic to primary/canary workload? if I recall correctly, we want to only modify the forward traffic actions, and keep "default forward 404" like action, isn't it?

Reply:
Thank you for your review. 

The code is designed to modify only the rules with 'forward' action types; all other rules will remain unchanged.

[e91ab8a](https://github.com/pipe-cd/pipecd/pull/4540/commits/e91ab8ac0f357a502c269a74d64097faf38add8b)

Reply:
@khanhtc1202